src/pages/Swap/index.tsx

```diff
@@ -314,10 +314,20 @@ export function Swap({
     [independentField, parsedAmount, showWrap, trade]
   )
 
-  const fiatValueInput = useUSDPrice(parsedAmounts[Field.INPUT], currencies[Field.INPUT] ?? undefined)
-  const fiatValueOutput = useUSDPrice(parsedAmounts[Field.OUTPUT], currencies[Field.OUTPUT] ?? undefined)
-  const showFiatValueInput = Boolean(parsedAmounts[Field.INPUT])
-  const showFiatValueOutput = Boolean(parsedAmounts[Field.OUTPUT])
+  const fiatValueInput = useUSDPrice(
+    parsedAmounts[Field.INPUT] ??
+      (currencies[Field.INPUT]
+        ? CurrencyAmount.fromRawAmount(currencies[Field.INPUT], JSBI.BigInt(10 ** currencies[Field.INPUT].decimals))
+        : undefined),
+    currencies[Field.INPUT]
+  )
+  const fiatValueOutput = useUSDPrice(
+    parsedAmounts[Field.OUTPUT] ??
+      (currencies[Field.OUTPUT]
+        ? CurrencyAmount.fromRawAmount(currencies[Field.OUTPUT], JSBI.BigInt(10 ** currencies[Field.OUTPUT].decimals))
+        : undefined),
+    currencies[Field.OUTPUT]
+  )
```

Comment:
this is muuuuch harder to parse quickly. Can you find a more readable way to break out the same logic?